src/python/pants/build_graph/address.py

```diff
@@ -181,8 +179,9 @@ def __init__(
         :param generated_base_target_name: If this Address refers to a generated subtarget, this
                                            stores the target_name of the original base target.
         """
-        self._spec_path = self.sanitize_path(spec_path)
+        self.validate_path(spec_path)
```

Comment:
Should probably remove the return value from this method too.

Reply:
Ah, I meant to do that. Thanks!